sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/Exceptions.scala

```diff
@@ -24,13 +28,17 @@ private object Exceptions {
   def isNotFoundException(throwable: Throwable): Boolean = {
     throwable match {
       case cosmosException: CosmosException =>
-        if (cosmosException.getStatusCode == HttpConstants.StatusCodes.NOTFOUND &&
-          cosmosException.getSubStatusCode == 0) {
-          true
-        } else {
-          false
-        }
+        isNotFoundExceptionCore(cosmosException)
       case _ => false
     }
   }
+
+  def isNotFoundExceptionCore(cosmosException: CosmosException): Boolean = {
+      if (cosmosException.getStatusCode == HttpConstants.StatusCodes.NOTFOUND &&
```

Comment:
nit: Can we just return the `if` clause? and remove the `if` and the `true` and `false`?

Reply:
Done